File: Lts/Under-progress/obylon-v7-clean/obylon-full/test_sig.go
package main

import (
	"fmt"
	"strings"
)

func canonicalSignPayload(payload map[string]interface{}, hardwareUUID string) []byte {
	var b strings.Builder
	b.WriteString("{")

	signFieldJSON(&b, payload, "license_id")
	b.WriteString(",")
	signFieldJSON(&b, payload, "node_id")
	b.WriteString(",")

	b.WriteString("hardware_uuid":)
	b.WriteString(" + hardwareUUID + ")
	b.WriteString(",")

	signFieldJSON(&b, payload, "expires_at")
	b.WriteString(",")
	signFieldJSON(&b, payload, "issued_at")
	b.WriteString(",")
	signFieldJSON(&b, payload, "status")

	b.WriteString("}")
	return []byte(b.String())
}

func signFieldJSON(b *strings.Builder, payload map[string]interface{}, key string) {
	b.WriteString(" + key + ":)
	val, ok := payload[key]
	if !ok || val == nil {
		b.WriteString("null")
		return
	}
	if strVal, isStr := val.(string); isStr {
		b.WriteString(" + strVal + ")
		return
	}
	b.WriteString("null")
}

func main() {
	payload := map[string]interface{}{
		"license_id": "456",
		"node_id":    "123",
		"expires_at": nil,
		"issued_at":  "2026-08-29T13:43:53.000Z",
		"status":     "active",
	}
	fmt.Println("GO sign_payload:", string(canonicalSignPayload(payload, "TEST-UUID")))
}

File: Lts/Under-progress/obylon-v7-clean/obylon-full/test_sig.py
import json
import base64
from nacl.signing import VerifyKey
import time

LICENSE_VERIFY_KEY_B64 = 'oQYy7eR/qxZOlKw/v9QNpmrcDWpNKGOx2YM0q++oXaY='
HARDWARE_UUID = 'TEST-UUID'

payload = {
    'license_id': '456',
    'node_id': '123',
    'expires_at': None,
    'issued_at': '2026-08-29T13:43:53.000Z',
    'status': 'active'
}

sign_payload = json.dumps({
    'license_id': payload.get('license_id'),
    'node_id': payload.get('node_id'),
    'hardware_uuid': HARDWARE_UUID,
    'expires_at': payload.get('expires_at'),
    'issued_at': payload.get('issued_at'),
    'status': payload.get('status')
}, separators=(',', ':')).encode('utf-8')

print('PYTHON sign_payload:', sign_payload)
